mqtt-impl/src/main/java/io/streamnative/pulsar/handlers/mqtt/utils/PulsarTopicUtils.java

```diff
@@ -72,7 +87,27 @@
         return promise;
     }
 
-    public static String getPulsarTopicName(String mqttTopicName) {
-        return StringUtils.strip(mqttTopicName, "/");
+    public static String getPulsarTopicName(String mqttTopicName, String defaultTenant, String defaultNamespace)
+            throws UnsupportedEncodingException {
+        if (mqttTopicName.startsWith(PERSISTENT_DOMAIN)
+                || mqttTopicName.startsWith(NON_PERSISTENT_DOMAIN)) {
+            List<String> parts = Splitter.on("://").limit(2).splitToList(mqttTopicName);
+            if (parts.size() < 2) {
+                throw new IllegalArgumentException("Invalid topic name: " + mqttTopicName);
+            }
+            String domain = parts.get(0);
+            String rest = parts.get(1);
+            parts = Splitter.on("/").limit(3).splitToList(rest);
+            if (parts.size() < 3) {
+                throw new IllegalArgumentException("Invalid topic name: " + mqttTopicName);
+            }
+            String tenant = parts.get(0);
+            String namespace = parts.get(1);
+            String localName = parts.get(2);
+            return TopicName.get(domain, tenant, namespace, URLEncoder.encode(localName, UTF8)).toString();
+        } else {
+            return TopicName.get(TopicDomain.persistent.value(), defaultTenant, defaultNamespace,
```

Comment:
Do we need to specify the topic domain?

Reply:
Yes, if users do not specify the domain name, MoP will use the `persistent` as default.

Reply:
Ok